src/static/js/cockpit.js

```diff
@@ -11,21 +11,47 @@
 
   //Cockpit is inheriting from EventEmitter2.
   var Cockpit = function Cockpit(csocket) {
-    this.socket = csocket;
+    var self = this;
+    this.uiLoader = new window.UiLoader();
+    this.rov = new window.MessageManager(csocket);
     this.sendUpdateEnabled = true;
     this.capabilities = 0;
     this.loadedPlugins = [];
-    this.loadPlugins();
-    console.log('loaded plugins');
-    // Register the various event handlers
-    this.listen();
+
+    this.loadUi(function() {
+      self.extensionPoints = {
```

Comment:
Note for a discussion about "at what level the cockpit webcontrol should exist at". Having the cockpit.js have a hardcode for where in the shadowdom it goes to expose the extension points seems like a leak of the abstraction that results in pulling the abstraction layer one level too high.